docs/guides/starter_guides/working_with_arrays.md

```diff
@@ -0,0 +1,478 @@
+Run the query below to see what our data looks like:
+
+```sql runnable
+-- SELECT
+-- *
+-- FROM ontime.ontime LIMIT 100
+
+SELECT
+    FlightDate,
+    Origin,
+    OriginCityName,
+    Dest,
+    DestCityName,
+    DepTime,
+    DepDelayMinutes,
+    ArrTime,
+    ArrDelayMinutes
+FROM ontime.ontime LIMIT 5
+```
+
+Let's take a look at the top 10 busiest airports in the US on a particular day chosen at random, say '2024-01-01'.
+We're interested in understanding how many flights depart from each airport.
+Our data contains one row per flight, but it would be convenient if we could group the data by the origin airport and roll the destinations into an array.
+
+To achieve this we can use the [`groupArray`](/sql-reference/aggregate-functions/reference/grouparray) aggregate function, which takes values of the specified column from each row and groups them in an array.
+
+Run the query below to see how it works:
+
+```sql runnable
+SELECT
+    FlightDate,
+    Origin,
+    groupArray(toStringCutToZero(Dest)) AS Destinations
+FROM ontime.ontime
+WHERE Origin IN ('ATL', 'ORD', 'DFW', 'DEN', 'LAX', 'JFK', 'LAS', 'CLT', 'SFO', 'SEA') AND FlightDate='2024-01-01'
+GROUP BY FlightDate, Origin
+ORDER BY length(Destinations)
+```
+
+The [`toStringCutToZero`](/sql-reference/functions/type-conversion-functions#tostringcuttozero) in the query above is used to remove null characters which appear after some of the airport's 3 letter designation.
+
+With the data in this format, we can easily find the order of the busiest airports by finding the length of the rolled up "Destinations" arrays:
+
+```sql runnable
+WITH
+    '2024-01-01' AS date,
+    busy_airports AS (
+    SELECT
+    FlightDate,
+    Origin,
+--highlight-next-line
+    groupArray(toStringCutToZero(Dest)) AS Destinations
+    FROM ontime.ontime
+    WHERE Origin IN ('ATL', 'ORD', 'DFW', 'DEN', 'LAX', 'JFK', 'LAS', 'CLT', 'SFO', 'SEA')
+    AND FlightDate = date
+    GROUP BY FlightDate, Origin
+    ORDER BY length(Destinations)
+    )
+SELECT
+    Origin,
+    length(Destinations) AS outward_flights
+FROM busy_airports
+ORDER BY outward_flights DESC
+```
+
+### arrayMap {#arraymap}
+
+We saw in the previous query that Denver International Airport was the airport with the most outward flights for our particular chosen day.
+Let's take a look at how many of those flights were on-time, delayed by 15-30 minutes or delayed by more than 30 minutes.
+
+Many of the array functions in ClickHouse are so-called ["higher-order functions"](/sql-reference/functions/overview#higher-order-functions) and accept a lambda function as the first parameter.
+The [`arrayMap`](/sql-reference/functions/array-functions#arrayMap) function is an example of one such higher-order function and returns a new array from the provided array by applying a lambda function to each element of the original array.
+
+Run the query below which uses the `arrayMap` function to see which flights were delayed or on-time:
+
+```sql
```

Comment:
We can do this without arrayMap I think:

```sql
SELECT
    Origin,
    toStringCutToZero(Dest) AS Destination,
    Tail_Number,
    DepDelayMinutes AS delay_minutes,
    if(DepDelayMinutes >= 30, 'DELAYED', 
      if(DepDelayMinutes >= 15, 'WARNING', 'ON-TIME')
    ) AS status
FROM ontime.ontime
WHERE Origin = 'DEN'
    AND FlightDate = '2024-01-01'
    AND DepTime IS NOT NULL
    AND DepDelayMinutes IS NOT NULL
ORDER BY DepDelayMinutes DESC
```

Let me think of an example that we can map

Reply:
And it uses `arrayZip` so get to show off another function. Even better!